insights/parsers/ld_library_path.py

```diff
@@ -1,48 +1,96 @@
 """
-LdLibraryPath - LD_LIBRARY_PATH of PIDs
-=======================================
+LdLibraryPath - LD_LIBRARY_PATH of Users
+========================================
 
-Parser for parsing the environment variable LD_LIBRARY_PATH of each PID.
+Parser for parsing the environment variable LD_LIBRARY_PATH of each user
 
 """
 
 from collections import namedtuple
 from insights import parser, Parser
+from insights.util import deprecated
 from insights.parsers import SkipException, ParseException
 from insights.specs import Specs
 
-LdLibraryPath = namedtuple('LdLibraryPath', ('pid', 'path', 'raw'))
-"""namedtuple: Type for storing the LdLibraryPath of PID"""
+LdLibraryPathU = namedtuple('LdLibraryPathU', ('path', 'raw'))
+"""namedtuple: Type for storing the LD_LIBRARY_PATH of users"""
 
 
-@parser(Specs.ld_library_path_of_pid)
-class PidLdLibraryPath(Parser, list):
+@parser(Specs.ld_library_path_of_user)
+class UserLdLibraryPath(Parser, list):
     """
-    Base class for parsing the ``LD_LIBRARY_PATH`` variable of each PID of the
-    system into a list.
+    Base class for parsing the ``LD_LIBRARY_PATH`` variable of each regular
+    user of the system into a list.
 
     Typical content looks like::
 
-        105901 /usr/sap/RH1/SYS/exe/run:/usr/sap/RH1/SYS/exe/uc/linuxx86_64:/sapdb/clients/RH1/lib
-        105902 /usr/sap/RH1/SYS/exe/uc/linuxx86_64:/usr/sap/RH1/SYS/exe/run
+        /usr/sap/RH1/SYS/exe/run:/usr/sap/RH1/SYS/exe/uc/linuxx86_64:/sapdb/clients/RH1/lib
+
+        /usr/sap/RH1/SYS/exe/uc/linuxx86_64:/usr/sap/RH1/SYS/exe/run
 
     Examples:
         >>> len(ld_lib_path)
-        2
+        3
         >>> isinstance(ld_lib_path[0].path, list)
         True
         >>> len(ld_lib_path[0].path)
         3
         >>> '/sapdb/clients/RH1/lib' in ld_lib_path[0].path
         True
-        >>> ld_lib_path[0].pid
-        '105901'
+        >>> '' in ld_lib_path[1].path  # The empty value is kept.
+        True
+
+    Raises:
+        SkipException: When the output is empty or nothing needs to parse.
+    """
+
+    def parse_content(self, content):
+        llds = []
+        for line in content:
+            raw = line
+            if line and line[0] == line[-1] and line[0] in ('\'', '"'):
+                line = line[1:-1]
+            llds.append(LdLibraryPathU(line.split(':'), raw))
+
+        if not llds:
+            raise SkipException("LD_LIBRARY_PATH not set.")
+
+        self.extend(llds)
+
+
+LdLibraryPath = namedtuple('LdLibraryPath', ('pid', 'path', 'raw'))
+"""
+.. warning::
+
+    This type is deprecated please use the :py:class:`LdLibraryPathU` instead.
+
+namedtuple: Type for storing the LdLibraryPath of PID
+"""
+
+
+class PidLdLibraryPath(Parser, list):
```

Comment:
@xiangce this parser didn't work and it was here for such a short period of time I would just remove this parser instead of deprecating it.